apps/performance/34-default-vs-onpush/src/app/person-list.component.ts

```diff
@@ -1,68 +1,53 @@
-import { Component, Input } from '@angular/core';
-
 import { CDFlashingDirective } from '@angular-challenges/shared/directives';
 import { CommonModule } from '@angular/common';
-import { FormsModule } from '@angular/forms';
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
 import { MatChipsModule } from '@angular/material/chips';
-import { MatFormFieldModule } from '@angular/material/form-field';
-import { MatInputModule } from '@angular/material/input';
 import { MatListModule } from '@angular/material/list';
+import { PersonComponent } from './person.component';
+import { SearchFieldComponent } from './search-field.component';
 
 @Component({
   selector: 'app-person-list',
+  changeDetection: ChangeDetectionStrategy.OnPush,
   standalone: true,
-  imports: [
-    CommonModule,
-    FormsModule,
-    MatListModule,
-    MatFormFieldModule,
-    MatInputModule,
-    MatChipsModule,
-    CDFlashingDirective,
-  ],
   template: `
     <h1 cd-flash class="text-center font-semibold" title="Title">
       {{ title | titlecase }}
     </h1>
 
-    <mat-form-field class="w-4/5" cd-flash>
-      <input
-        placeholder="Add one member to the list"
-        matInput
-        type="text"
-        [(ngModel)]="label"
-        (keydown)="handleKey($event)" />
-    </mat-form-field>
+    <app-search-field (searchFieldOutput)="addName($event)"></app-search-field>
 
     <mat-list class="flex w-full">
       <div *ngIf="names?.length === 0" class="empty-list-label">Empty list</div>
-      <mat-list-item
-        *ngFor="let name of names"
-        cd-flash
-        class="text-orange-500">
-        <div MatListItemLine class="flex justify-between">
-          <h3 title="Name">
-            {{ name }}
-          </h3>
-        </div>
-      </mat-list-item>
+      <app-person
+        [name]="name"
+        *ngFor="let name of names; let index = index; trackBy: userByName" />
       <mat-divider *ngIf="names?.length !== 0"></mat-divider>
     </mat-list>
   `,
   host: {
     class: 'w-full flex flex-col items-center',
   },
+  imports: [
+    CommonModule,
+    MatListModule,
+    MatChipsModule,
+    CDFlashingDirective,
+    SearchFieldComponent,
+    PersonComponent,
+  ],
 })
 export class PersonListComponent {
   @Input() names: string[] = [];
   @Input() title = '';
+  public index = 0;
```

Comment:
nit: if you want, you can refactor using all new angular features: signal, signal inputs, control flow...